maliput_ros/include/maliput_ros/ros/maliput_query.h

```diff
@@ -0,0 +1,61 @@
+// BSD 3-Clause License
+//
+// Copyright (c) 2022, Woven Planet.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// * Neither the name of the copyright holder nor the names of its
+//   contributors may be used to endorse or promote products derived from
+//   this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#pragma once
+
+#include <memory>
+#include <utility>
+
+#include <maliput/api/road_geometry.h>
+#include <maliput/api/road_network.h>
+#include <maliput/common/maliput_throw.h>
+
+namespace maliput_ros {
+namespace ros {
+
+/// @brief Proxy to a maliput::api::RoadNetwork to make queries to it.
+class MaliputQuery final {
+ public:
+  /// @brief Constructs a MaliputQuery.
+  /// @param[in] road_network The maliput::api::RoadNetwork to hold. It must not be nullptr.
+  /// @throws maliput::common::assertion_error When @p road_network is nullptr.
+  explicit MaliputQuery(std::unique_ptr<maliput::api::RoadNetwork> road_network)
+      : road_network_(std::move(road_network)) {
```

Comment:
QQ: Is there a reason why MaliputQuery is the owner of the RoadNetwork?
I am ok with this, at the moment there are no other applications

Reply:
Simply because there is no other one requiring to hold the ownership of it. The ownership lifecycle will be tied as well so even though you are right, there is no explicit ownership relationship, it sounded OK to me leave to the MaliputQuery the responsibility to own that memory and simplify the management at the node level.